src/data/userData.ts

```diff
@@ -86,8 +91,14 @@ export async function updateUser(
 export async function deleteUser(id: string): Promise<void> {
   try {
     const docRef = doc(db, 'users', id);
+    const snapshot = await getDoc(docRef);
+    if (!snapshot.exists()) {
+      throw new Error('User does not exist');
```

Comment:
Throwing error here just get's caught in error creation in catch statement causing a nested error like the following to be logged: 
`Failed to delete user: Error: Failed to delete user: Error: User does not exist`

it shouldn't be nested. You try to just have it return null if it doesn't exist and check if the returned value is null in the delete.ts file where the function is called

Reply:
Got it, that makes sense